jerry-core/jerry.c

```diff
@@ -2079,64 +2047,56 @@ jerry_parse_and_save_snapshot (const jerry_api_char_t *source_p, /**< script sou
     ecma_deref_object (error_obj_p);
   }
 
-  snapshot_report_byte_code_compilation = false;
-
-  if (parse_status == JSP_STATUS_OK
-      && !snapshot_error_occured)
+  if (parse_status != JSP_STATUS_OK)
   {
-    JERRY_ASSERT (snapshot_last_compiled_code_offset != 0);
-
-    jerry_snapshot_header_t header;
-    header.last_compiled_code_offset = (uint32_t) snapshot_last_compiled_code_offset;
-    header.is_run_global = is_for_global;
-
-    size_t compiled_code_size = snapshot_buffer_write_offset - compiled_code_start;
+    return 0;
+  }
 
-    lit_mem_to_snapshot_id_map_entry_t *lit_map_p = NULL;
-    uint32_t literals_num;
+  snapshot_add_compiled_code (bytecode_data_p);
 
-    if (!lit_dump_literals_for_snapshot (buffer_p,
-                                         buffer_size,
-                                         &snapshot_buffer_write_offset,
-                                         &lit_map_p,
-                                         &literals_num,
-                                         &header.lit_table_size))
-    {
-      JERRY_ASSERT (lit_map_p == NULL);
-      snapshot_buffer_write_offset = 0;
-    }
-    else
-    {
-      jerry_snapshot_set_offsets (buffer_p + compiled_code_start,
-                                  (uint32_t) compiled_code_size,
-                                  lit_map_p);
+  if (snapshot_error_occured)
+  {
+    return 0;
+  }
 
-      jrt_write_to_buffer_by_offset (buffer_p,
-                                     buffer_size,
-                                     &header_offset,
-                                     &header,
-                                     sizeof (header));
+  jerry_snapshot_header_t header;
+  header.version = JERRY_SNAPSHOT_VERSION;
```

Comment:
It's fine if this is just not acceptable style for this project. I was only pointing it out because I recently had to fix a pretty nasty regression where a field was added to a struct that was initialized field by field as you do here and ended up uninitialized:

```
typedef struct {
  int a;
  int b;
  int c;
} foo_t;
...
foo_t bar;
// when this code was written, foo_t only had two fields, a and b.
bar.a = 1;
bar.b = 2;
...
// bar.c is uninitialized
```